src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -56,108 +71,109 @@ private class CacheItem
         };
 
         /// <summary>Default maximum number of transactions in the cache.</summary>
-        public const int CacheMaxItemsDefault = 100000;
+        public const int CacheMaxItemsDefault = 100_000;
+
+        /// <summary>The current batch size in bytes.</summary>
+        private long currentBatchSizeBytes;
+
+        /// <summary>Maximum interval between saving batches.</summary>
+        /// <remarks>Interval value is a prime number that wasn't used as an interval in any other component. That prevents having CPU consumption spikes.</remarks>
+        private const int BatchMaxSaveIntervalSeconds = 41;
 
-        /// <summary>Length of the coinview cache flushing interval in seconds.</summary>
-        /// <seealso cref="lastCacheFlushTime"/>
-        public const int CacheFlushTimeIntervalSeconds = 60;
+        /// <summary>Maximum number of bytes the batch can hold until the rewind data items are stored to the disk.</summary>
+        internal const int BatchThresholdSizeBytes = 5 * 1_000 * 1_000;
 
         /// <summary>Instance logger.</summary>
         private readonly ILogger logger;
 
+        /// <summary>Queue which contains rewind data items that should be saved to the database.</summary>
+        private readonly AsyncQueue<QueuedRewindData> rewindDataQueue;
```

Comment:
This is a bit alarming to me, why do we need an asyncqueue in the utxo store layer?